File: performance_langage/performanceLangage.py
import time

def now():
    return time.perf_counter()*1000
temps_initial =now()

def add(a, b):
    return a+b
print(add(5, 10))

def add_array(arr, n):
    return [x + n for x in arr]
print(add_array([3, 4, 1, 10], 1))

def factorielle(n):
    if n <= 1:
        return 1
    else:
        return n * factorielle(n - 1)
    print(factorielle(5))

temps_final = now()
temps_ecoule = temps_final - temps_initial
print(f"L'exécution a duré {temps_ecoule:.4f} millisecondes")

File: performance_langage/performanceLangage.js
function now() {
    return performance.now(); 
  }


function add(n1, n2){
    return n1 + n2;
  }  

function add_array(arr, n)  {
    return arr.map(x => x + n);
  };
  console.log(add_array([3, 4, 1, 10], 1));

 
function factorielle(n){
    return (n != 1) ? n * factorielle(n - 1) : 1;  
}
function mesurerExecution(nom, fonction, ...args) {
    let debut = now();
    let resultat = fonction(...args);
    let fin = now();
    console.log(`${nom}(${args}) = ${resultat}`);
    console.log(`Durée : ${(fin - debut).toFixed(4)} ms\n`)
}
  
mesurerExecution("add", add, 5, 10);
mesurerExecution("add_array", add_array, [3, 4, 1, 10], 1);
mesurerExecution("factorielle", factorielle, 5);
  // La comparaison visuelle montre que python est beaucoup plus rapide que javaScript

  